text/0000-natvis.md

```diff
@@ -0,0 +1,428 @@
+- Feature Name: `natvis`
+- Start Date: 2021-11-01
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+This RFC aims to improve the debugging experience for Rust developers, by
+enabling Rust developers to package debugger visualization scripts with their
+crates.
+
+# Motivation
+[motivation]: #motivation
+
+Most, if not all, Rust developers will at some point have to debug an issue
+in their crate. Trying to view types as they are laid out in memory is not
+always the most telling. Furthermore when viewing types from external crates,
+the information is even harder to interpret. 
+
+Many languages and debuggers enable developers to control how a type is
+displayed in a debugger. These are called "debugger visualizations" or "debugger
+views". Debugger views are merely a convenience for some types, such as
+`Vec<T>`, but are essential for types such as `HashMap<T>`, where non-trivial
+logic is needed in order to correctly display the contents of a type.
+
+Currently, Rust provides visualizations for a handful of types defined in its
+standard library via `.natvis` files or python scripts. However, this support
+is inflexible; updating it requires modifying the Rust toolchain itself,
+and either using a local build of the toolchain or waiting for a new upstream
+build of the toolchain. It is not feasible for developers of ordinary crates
+to update the Rust toolchain, solely to add visualizations for their crates.
```

Comment:
It would be useful to explain the platform constraints of natvis files here, I think. Perhaps also give an example of what something like HashMap looks like with and without the visualisation applied (put another way, this RFC is assuming that the reader has some knowledge of debugger implementation details, but would be improved if it only assumed the reader has used a debugger)